### PR TITLE
Create calculatePoseFromTarget

### PR DESCRIPTION
**Pull request by alicekuznetsov**
Reasons why this _might_ not work:
- result.getTimestampSeconds(); is not synced with Timer.getFPGATimestamp();
- I'm not smart
- Also, chasing the target won't work if we're constantly resetting our position based on the target we're chasing, so remember that ig

### CODE REVIEW

**Comment by IanSzalai** [comment]
I'm not really sure how `result.getTimestampSeconds()` differs from `Timer.getFPGATimestamp()` but worse case we can purely use the former and it should be fine.

**Comment by IanSzalai** [comment]
Ideally, the vision subsystem has the bulk of the vision processing code, rather than a command. This would distill into some `getRobotPose()` (or similarly named) method that is used in the `UpdatePoseEstimator` command. To my understanding, what complicates this is that if there is no valid pose, you don't want to add a vision measurement. I see two ways around this.

1. Simple include another getter method in the vision system that will simply return true if the vision data is good to use.
```java
if (subVision.isPoseValid()) {
  addVisionMeasurement(subVision.getRobotPose());
}
// method names for example
```
2. Return null from `subVision.getRobotPose()` if the data isn't good to use. Then you could either check for null in `UpdatePoseEstimator` or in `subDrivetrain.addVisionMeasurement()` eg
```java
public void addVisionMeasurement(Pose2d visionRobotPose, double timestampSeconds) {
  if (visionRobotPose != null) {
    poseEstimator.addVisionMeasurement(
        visionRobotPose,
        timestampSeconds);
  }
}
```
or
```java
// execute method of UpdatePoseEstimator
public void execute() {
  subDrivetrain.updatePoseEstimator();
  if (subVision.getRobotPose() != null) {
    subDrivetrain.addVisionMeasurement(subVision.getRobotPose(), subVision.getRobotPoseTimestamp());
  }
}
```

I may be misunderstanding the problem at hand, please let me know if that's the case.

File: src/main/java/frc/robot/commands/getInRangeOfTag.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
 import frc.robot.subsystems.Vision;
 
 // Note: This command assumes that you know where you are on the field, since every other position is relative to you
+// Ok so I kinda wanna throw this command into a fire and start over so I understand it better B)
+// Also it would just be better and I kinda need to anyway
```
Comment (IanSzalai) [inline src/main/java/frc/robot/commands/getInRangeOfTag.java]: i agree

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -285,10 +285,10 @@ public void updatePoseEstimator() {
    * 
    * @param visionRobotPose Pose of robot as calculated by the vision system
    */
-  public void addVisionMeasurement(Pose2d visionRobotPose) {
+  public void addVisionMeasurement(Pose2d visionRobotPose, double timestampSeconds) {
     poseEstimator.addVisionMeasurement(
         visionRobotPose,
-        Timer.getFPGATimestamp());
+        timestampSeconds);
```
Comment (IanSzalai) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:296]: assuming you're only calling this method with valid pose data, does this change only matter for accuracy beyond +/- 20ms (50hz)?

Comment (alicekuznetsov) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:296]: I just thought that a timestamp associated with the vision data measurement would be more accurate than a timestamp that's run slightly after the vision data. I don't think the increase in accuracy is too influential, but I think it's still better to do it like this.

Comment (IanSzalai) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:296]: i agree. the theoretical worst delta is 20ms (if i understand this correctly) and that could cause noticeable error

**Comment by IanSzalai** [comment]
new changes look good to me. do you think that this structure makes the most sense?

**Comment by alicekuznetsov** [comment]
> new changes look good to me. do you think that this structure makes the most sense?

I think so. Combining all of the types of pose estimators into one command but keeping them separate in subsystems seems smart. 
Not having real positions for each apriltag (ie. every tag is at 0 rn) is going to make the whole process a bit weird, but that sounds like a problem for future me 

Just committed changes that should (?) fix the "chasing tag" problem. Correct me if we shouldn't use it or if I did it wrong. If we decide to implement a chasing tag code, I will also add a filter on the specific tag that we use to make its pose invalid.

**Pull request merged**